src/traits.rs

```diff
@@ -58,6 +58,67 @@ pub trait Context {
         hostcalls::enqueue_shared_queue(queue_id, value)
     }
 
+    /// Sends HTTP request with serialized `headers`, `body`, and serialized trailers to upstream.
+    ///
+    /// `on_http_call_response()` will be called with `token_id` when the response is received by the host, or after the timeout.
+    ///
+    /// # Arguments
+    ///
+    /// * `upstream` - The name of the upstream to send the request to (in your envoy configuration)
+    /// * `headers` - The headers to send with the request (in the form of `Vec<(&str, &str)>`)
+    /// * `body` - The body of the request (in the form of `Option<&[u8]>` - `None` if no body)
+    /// * `trailers` - The trailers to send with the request (in the form of `Vec<(&str, &str)>`)
+    /// * `timeout` - The timeout for the request
+    ///
+    /// # Returns
+    ///
+    /// * `OK` on success.
+    /// * `BAD_ARGUMENT` for unknown upstream, or when headers are missing required `:authority`, `:method` and/or `:path` values.
+    /// * `INTERNAL_FAILURE' when the host failed to send requested HTTP call.
+    /// * `INVALID_MEMORY_ACCESS` when `upstream_data`, `upstream_size`, `headers_data`, `headers_size`, `body_data`, `body_size`, `trailers_data`, `trailers_size` and/or `return_call_id` point to invalid memory address.
```

Comment:
It looks that you've copied those return values from the spec, which is generally fine... However, this SDK:
- combines `*_size` and `*_data` into vectors and strings,
- assumes that the memory objects are correct, so `INVALID_MEMORY_ACCESS` cannot happen and if it does, then it will result in `panic`,
- it's opinionated about which return values are allowed, so if you want to document them, then take a look at `src/hostcalls.rs` and see which return values are there.

Reply:
I removed the docs from the spec and wrote something myself. Is that better?

Reply:
I didn't mean that including text from the (now merged) spec is wrong. I meant that it had to be adjusted to account for the higher-level types used in the SDK.

I'll take a look at it in the next few days, thanks!